File: api.ml/gateway/src/main/java/com/fluxcapacitor/api/gateway/filters/route/CanaryRoutingFilter.java
package com.fluxcapacitor.api.gateway.filters.route;

import com.netflix.zuul.ZuulFilter;
//import org.springframework.web.servlet.support.RequestContext;
//import javax.servlet.http.HttpServletRequest;
//
//  http://cloud.spring.io/spring-cloud-netflix/spring-cloud-netflix.html
//
public class CanaryRoutingFilter extends ZuulFilter {  
    @Override
    public String filterType() {
        return "post";
    }

    @Override
    public int filterOrder() {
        return 1;
    }
    
    @Override
    public boolean shouldFilter() {
        return true;
    }

    @Override
    public Object run() {
        System.out.println(String.format("%s request to %s", "foo", "bar"));
        
        return null;
    }
}

File: api.ml/gateway/src/main/scala/com/fluxcapacitor/api/gateway/filters/route/CanaryRoutingFilter.scala
package com.fluxcapacitor.api.gateway.filters.route

import com.netflix.zuul.ZuulFilter
//
//  http://cloud.spring.io/spring-cloud-netflix/spring-cloud-netflix.html
//
class CanaryRoutingFilter extends ZuulFilter {  
    @Override
    def filterType(): String = {
      "post"
    }

    @Override
    def filterOrder(): Int = {
      1
    }
    
    @Override
    def shouldFilter(): Boolean = {
      true
    }

    @Override
    def run(): Object = {
      System.out.println(s"""${"foo"} request to ${"bar"}""");
        
      null
    }
}
